packages/amplify-category-auth/index.js

```diff
@@ -188,19 +188,29 @@ async function checkRequirements(requirements, context) {
 
 async function initEnv(context) {
   const { amplify } = context;
-  const { resourcesToBeCreated, resourcesToBeDeleted } = await amplify.getResourceStatus('auth');
+  const { resourcesToBeCreated, resourcesToBeDeleted, resourcesToBeUpdated } = await amplify.getResourceStatus('auth');
 
   resourcesToBeDeleted.forEach((authResource) => {
     amplify.removeResourceParameters(context, 'auth', authResource.resourceName);
   });
 
-  const authTasks = resourcesToBeCreated.map((authResource) => {
+  let authTasks = resourcesToBeCreated.map((authResource) => {
     const { resourceName } = authResource;
     return async () => {
       const config = await updateConfigOnEnvInit(context, 'auth', resourceName);
       context.amplify.saveEnvResourceParameters(context, 'auth', resourceName, config);
     };
   });
+
+  const updateTasks = resourcesToBeUpdated.map((authResource) => {
+    const { resourceName } = authResource;
+    return async () => {
+      const config = await updateConfigOnEnvInit(context, 'auth', resourceName);
+      context.amplify.saveEnvResourceParameters(context, 'auth', resourceName, config);
```

Comment:
Can we extract this out and do it as single step instead of doing it once for create and once for update